rc4me/fetch_rcs.py

```diff
@@ -0,0 +1,88 @@
+"""Clone dotfile repo to rc4me home directory."""
+
+from pathlib import Path
+import logging
+import click
+import git
+
+logging.basicConfig(level=logging.DEBUG)
+logger = logging.getLogger(__name__)
+
+
+@click.command()
+@click.argument("repo", type=str)
+def _fetch_rcs_cli(rc4me_home: Path, repo: str) -> None:
+    """CLI wrapper for fetch_rcs function.
+
+    Args:
+        repo: Local repo or GitHub repo stub (e.g. "jeffmm/vimrc").
+
+    Returns:
+        None
+
+    """
+    fetch_rcs(rc4me_home, repo)
+
+
+def fetch_rcs(rc4me_home: Path, repo: str) -> None:
+    """Clone RC repository to local directory.
+
+    Clones rc4me repository to rc4me home directory at $HOME/.rc4me. If the
+    repo already exists locally, fetch updates and confirm update to origin
+    master. If the .rc4me home directory does not exist, scaffold it.
+
+    Args:
+        repo: Local repo or GitHub repo stub (e.g. "jeffmm/vimrc").
+
+    Returns:
+        None
+
+    """
+    repo_dir = rc4me_home / repo
+    # Check if the repo is local first
+    if not rc4me_home.exists():
+        init_rc4me(rc4me_home)
+    if repo_dir.exists():
+        # If the repo exists, fetch and update
+        fetch_info = git.Repo(repo_dir).remote("origin").fetch()
+        # Check that the repo is up to date
+        if fetch_info[0].commit.hexsha != git.Repo(repo_dir).head.commit.hexsha:
+            # Update on user confirmation
+            if _check_if_overwrite(repo):
+                git.Repo(repo_dir).remote("origin").pull("master")
+    else:
+        # Otherwise, clone the repo
+        if Path(repo).exists():
```

Comment:
How can `Path(repo).exist()` if `repo_dir` doesn't exist?

Reply:
I'm a little confused by this `if`/`else`

Reply:
Yeah, I will comment this better for clarity. This setup allows the user to target a repo on their local machine. The first `if repo_dir.exists()` checks if the target repo already exists inside the `rc4me_home` directory. If it doesn't, it checks first whether a local directory exists with the target repo name (`if Path(repo).exists()`), and if so, clones that. If neither of those are true, then it assumes you are referencing a git repo. And I don't know why this isn't an `if/elif/else` statement, so I'll fix that too.